docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +278,229 @@ Note that:
+    * The `url` conveys the fact the `c8y_configuration_plugin` expects that, on transfer success,
+      the file will be stored `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshot/$TYPE`.
+      Note that when the `$TYPE` is not explicitly defined in `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`,
+      then the `$PATH` is used as the configuration type.
+1. On reception of a configuration request on the topic `tedge/$CHILD_DEVICE_ID/commands/req/config_snapshot`,
+   The child-device specific software for configuration management:
+   1. Uses the `path` and `type` information to retrieve the requested configuration content.
+   Note that these pieces of information are provided by the child-device specific software itself,
+   and make sense only in the specific context of the device operating system and software.
+   1. `PUT`s the content to the `url` specified by the request message.
+1. The child-device specific software for configuration management,
+   notifies thin-edge over MQTT about the progress of the configuration snapshot request.
+    1. These messages are published on the topic
+       `tedge/$CHILD_DEVICE_ID/commands/res/config_snapshot`
+    1. There is three different payloads to notify the configuration operation started,
+       succeed or failed for some reason.
+    1. All these payloads are JSON records with 3 required fields:
+        * `"status": "$STATUS"` where the status is either "executing", "successful" or "failed"
+        * `"path": "$PATH"`
+        * `"type": "$TYPE"`
+        * `"reason": "$ERROR_MSG"` telling the cause of the error if any.
+    1. The child-device configuration software must send at least a success or an error message,
+       depending on the success of the `PUT` operation.
+       It should also send an executing message before starting to process the request.
+1. On reception an operation status message,
+   the `c8y_configuration_plugin` notifies the cloud accordingly.
+    1. When a success message is received,
+       then the configuration plugin transfers to the cloud the content `PUT` by the child-device
+       under `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/$TYPE` and
+       finally removes this file when acknowledged by the cloud.
+    1. If no responses are received from the child devices after 60s,
+       then a timeout error is sent to the cloud.
+    1. If a notification message is received while none is expected for this type of configuration
+       (this might notably arrive after a timeout), then this notification message is ignored.
+       
+### The child device uploads its configuration file list on start and on change
+
+The configuration files for a child device `$CHILD_DEVICE_ID` are defined by
+the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+The content of this file is provided by the child-device configuration software
+but must be known by the `c8y_configuration_plugin`
```

Comment:
It is not necessarily provided by the child-device. Why not allow a user to create that file manually, for child-device that cannot provided that information (same as for the main device)?

Reply:
What would prevent to have an admin this configuration file? What I meant here is that the `c8y-configuration-plugin`  is not defined on the main device but on the child device (or where is running the child device specific software if a proxy is required).

Reply:
Ok, understood.